### PR TITLE
Bug 1156907 - migrate counters from graph server into treeherder

### PR DESCRIPTION
**Pull request by jmaher**
This creates signatures and series for all metrics that we have in the talos_counters part of the talosdata json object.  In addition we gate on ensuring the values are single numbers (i.e. not lists, objects) as a preventative measure.

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/558)

### CODE REVIEW

**Comment by wlach** [comment]
Also, this is failing travis because of minor flake8 problems. I'd suggest running the tests (which include this check) before submitting in the future:

http://treeherder.readthedocs.org/common_tasks.html#running-the-tests

**Comment by jmaher** [comment]
- there is one issue with it on xperf counters- I will fix the flakes
   stuff as well.

**Comment by jmaher** [comment]
updated with local tests passing and verified xperf counters locally!

File: treeherder/etl/perf_data_adapters.py [hunk 1]
```diff
@@ -235,6 +234,48 @@ def adapt_and_load(self, reference_data, job_data, datum):
...
+                    "total_replicates": 1,
+                    "min": 0,
+                    "max": 0,
+                    "mean": 0,
+                    "std": 0,
+                    "median": 0
+                }
+                for stat in ['min', 'max', 'mean', 'median', 'std', 'total_replicates']:
+                    if stat in talos_datum["talos_counters"][_test]:
+                        # in case we have non int/float data, lets ignore it
```
Comment (wlach) [inline treeherder/etl/perf_data_adapters.py]: Should we maybe at least log a warning for this case?

File: treeherder/etl/perf_data_adapters.py [hunk 2]
```diff
@@ -235,6 +234,48 @@ def adapt_and_load(self, reference_data, job_data, datum):
             result_set_id = job_data[_job_guid]['result_set_id']
             push_timestamp = job_data[_job_guid]['push_timestamp']
 
+            # counters will not be part of the summary series
+            # counters have a json obj {'stat': val} instead of [val1, val2, ...]
+            for _test in talos_datum["talos_counters"].keys():
```
Comment (wlach) [inline treeherder/etl/perf_data_adapters.py]: It looks like you're assuming that the talos object always has a talos_counters key, which is not necessarily the case. This is causing the unit tests to fail.

That brings up the point that (1) the unit test sample data should be updated to incorporate some counters, so we're testing that and (2) we should only process talos_counters if that data actually exists

File: treeherder/etl/perf_data_adapters.py [hunk 3]
```diff
@@ -268,36 +309,37 @@ def adapt_and_load(self, reference_data, job_data, datum):
...
+                                              summary_properties,
+                                              summary_data)
+                # FIXME: this stuff is pretty big and it's unclear how useful it
+                # might be. consider adding it at some point in the future
+                # (see: https://bugzilla.mozilla.org/show_bug.cgi?id=1142631)
+                # for key in ['test_aux', 'talos_aux', 'results_aux',
+                #             'results_xperf']:
+                #     aux_blob = talos_datum.get(key)
+                #     if aux_blob:
+                #         obj['blob']['metadata'][key] = aux_blob
```
Comment (wlach) [inline treeherder/etl/perf_data_adapters.py]: Let's just remove this comment block-- we've decided what to do with this data. :)

File: treeherder/etl/perf_data_adapters.py (hunk 2 above)
Comment (jmaher) [inline treeherder/etl/perf_data_adapters.py]: my latest code should have check for that, maybe it didn't get pushed?  I will fix it with the other items.

File: treeherder/etl/perf_data_adapters.py (hunk 1 above)
Comment (jmaher) [inline treeherder/etl/perf_data_adapters.py]: if we log a warning how would we do that, where would it be seen?

Comment (wlach) [inline treeherder/etl/perf_data_adapters.py]: It's pretty easy, we don't do it in this file, but we do elsewhere. Just import logging, get a logger object, and then use it as needed. See e.g.:

https://github.com/mozilla/treeherder/blob/master/treeherder/etl/oauth_utils.py#L11
https://github.com/mozilla/treeherder/blob/master/treeherder/etl/oauth_utils.py#L61

The warning would appear in the treeherder log (/var/log/treeherder/treeherder.log). Or once we switch to heroku, it'll be in papertrail (https://papertrailapp.com/). Useful for troubleshooting!

**Comment by jmaher** [comment]
updated the PR with all feedback

File: tests/etl/test_perf_data_adapters.py [hunk 4]
```diff
@@ -41,6 +41,10 @@ def test_adapt_and_load():
         # one extra result for the summary series
         result_count += (len(datum['blob']["results"]) + 1)
 
+        # we create one per counter
+        if 'talos_counters' in datum['blob']:
+            result_count += (len(datum['blob']["talos_counters"]))
```
Comment (wlach) [inline tests/etl/test_perf_data_adapters.py]: No need for the parens around len(...)

File: treeherder/etl/perf_data_adapters.py [hunk 5]
```diff
@@ -235,6 +237,54 @@ def adapt_and_load(self, reference_data, job_data, datum):
...
+                        "std": 0,
+                        "median": 0
+                    }
+                    for stat in ['min', 'max', 'mean', 'median', 'std', 'total_replicates']:
+                        if stat in talos_datum["talos_counters"][_test]:
+                            # in case we have non int/float data, lets ignore it
+                            try:
+                                series_data[stat] = float(talos_datum["talos_counters"][_test][stat])
+                            except:
+                                logger.info("Talos counters for job %s, "
```
Comment (wlach) [inline treeherder/etl/perf_data_adapters.py]: This should be logger.warning, IMO

File: treeherder/etl/perf_data_adapters.py [hunk 6]
```diff
@@ -268,36 +318,29 @@ def adapt_and_load(self, reference_data, job_data, datum):
...
-            #             'results_xperf']:
-            #     aux_blob = talos_datum.get(key)
-            #     if aux_blob:
-            #         obj['blob']['metadata'][key] = aux_blob
-
-            validate(obj, self.treeherder_perf_test_schema)
-            self._add_performance_artifact(job_id, summary_signature,
-                                           summary_properties, obj,
-                                           _name, 'summary', summary_data)
+            if subtest_signatures != []:
```
Comment (wlach) [inline treeherder/etl/perf_data_adapters.py]: `if subtest_signatures:` is sufficient here

File: tests/etl/test_perf_data_adapters.py (hunk 4 above)
Comment (wlach) [inline tests/etl/test_perf_data_adapters.py]: Oops, I guess it was like this before. Might as well change it while you're there though. :)

File: tests/etl/test_perf_data_adapters.py [hunk 7]
```diff
@@ -41,6 +41,10 @@ def test_adapt_and_load():
         # one extra result for the summary series
         result_count += (len(datum['blob']["results"]) + 1)
 
+        # we create one per counter
```
Comment (wlach) [inline tests/etl/test_perf_data_adapters.py]: --> # we create one performance series per counter

File: treeherder/etl/perf_data_adapters.py (hunk 6 above)
Comment (wlach) [inline treeherder/etl/perf_data_adapters.py]: Actually, wait, on second thought this is a good change but we probably shouldn't include it in this patch.

Let's fix this as part of bug 1162522. We'll want to update the database to remove the relevant series after applying this change.

Comment (wlach) [inline treeherder/etl/perf_data_adapters.py]: Disregard this comment, I misunderstood the code change here.

**Pull request merged**